### PR TITLE
[MIG][14.0] account_invoice_force_number

### PR DESCRIPTION
**Pull request by SimoneVagile**

### CODE REVIEW

**Review by chusamo — APPROVED** [review]
LGTM

**Review by fshah-initos — APPROVED** [review]
Code review and Functional Test. LGTM

**Comment by fshah-initos** [comment]
Someone can merge the PR hence it is approved.

**Comment by fshah-initos** [comment]
@all Can somone merge this PR hence it is approved?

File: account_invoice_force_number/tests/test_account_force_number.py [hunk 1]
```diff
@@ -4,49 +4,43 @@
 
 
 from odoo.exceptions import UserError
-from odoo.tests.common import TransactionCase
 
+from odoo.addons.account.tests.common import AccountTestInvoicingCommon
+
+
+class TestAccountForceNumber(AccountTestInvoicingCommon):
+    def create_invoice(self, invoice_amount=450, move_name=None):
```
Comment (tafaRU) [inline account_invoice_force_number/tests/test_account_force_number.py]: Considering that in your tests you always use the default value of `invoice_amount`, I would avoid this parameter.

File: account_invoice_force_number/tests/test_account_force_number.py [hunk 2]
```diff
@@ -4,49 +4,43 @@
...
+                        "quantity": 1,
+                        "price_unit": invoice_amount,
+                        "tax_ids": [(6, 0, [])],
+                    },
+                )
+            ],
+        }
+        invoice = (
+            self.env["account.move"]
+            .with_context(default_move_type=type)
```
Comment (tafaRU) [inline account_invoice_force_number/tests/test_account_force_number.py]: Typo: s/`type`/`"out_invoice"`

File: account_invoice_force_number/tests/test_account_force_number.py [hunk 3]
```diff
@@ -4,49 +4,43 @@
...
+            .create(invoice_vals)
+        )
+        if move_name is not None:
```
Comment (tafaRU) [inline account_invoice_force_number/tests/test_account_force_number.py]: Simply check it by using: `if move_name`.

**Review by tafaRU — CHANGES_REQUESTED** [review]

**Comment by SimoneVagile** [comment]
@tafaRU thanks for review. All asked changes has done.

File: account_invoice_force_number/tests/test_account_force_number.py [hunk 4]
```diff
@@ -4,49 +4,43 @@
...
 
 from odoo.exceptions import UserError
-from odoo.tests.common import TransactionCase
 
+from odoo.addons.account.tests.common import AccountTestInvoicingCommon
+
+
+class TestAccountForceNumber(AccountTestInvoicingCommon):
+    def create_invoice(self, move_name=None):
+        partner_id = self.env.ref("base.res_partner_12")
```
Comment (tafaRU) [inline account_invoice_force_number/tests/test_account_force_number.py]: Even though the invoice creation doesn't raise any error, I'd prefer to get assignment compliant with the variable name:
`partner_id = self.env.ref("base.res_partner_12").id`

**Review by tafaRU — CHANGES_REQUESTED** [review]
Sorry, I missed this in my previous review. Could you modify it?

File: account_invoice_force_number/tests/test_account_force_number.py (hunk 4 above)
Comment (SimoneVagile) [reply account_invoice_force_number/tests/test_account_force_number.py]: No problem, now it's correct :ok_hand:

**Review by tafaRU — APPROVED** [review]
LGTM code and functional review (on runbo) both

**Comment by kittiu** [comment]
@SimoneVagile seem there is a bug on v14 -> https://github.com/OCA/account-invoicing/issues/931

**Pull request merged**